.github/workflows/sdk-ci.yml

```diff
@@ -0,0 +1,95 @@
+name: dataprotector SDK - CI
+
+on:
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'packages/sdk/**'
+
+concurrency:
+  group: ${{ github.ref }}-sdk-ci
+  cancel-in-progress: true
+
+jobs:
+  check-code:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install Dependencies
+        working-directory: packages/sdk
+        run: |
+          npm ci
+          npm run codegen
+
+      - name: Format
+        working-directory: packages/sdk
+        run: npm run lint
+
+      - name: Lint
+        working-directory: packages/sdk
+        run: npm run check-format
+
+      - name: Check types
+        working-directory: packages/sdk
+        run: npm run check-types
+
+  test:
```

Comment:
Why not run everything on in the same job to avoid re-installing deps each time?

Reply:
Jobs are parallelized; at the end of the day, this will save time waiting for the CI.
Also, separating job concerns will let us know when a job is broken, regardless of the others' status.